### PR TITLE
libc: add common machine atomic operation

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
## Summary
Add atomic operation in libc matchine.
Witch could be used by all arch code.
and remove atomic code in riscv.

## Impact
No

## Testing
testing on arch xtesna

### CODE REVIEW

**Comment by Ouss4** [comment]
It looks like we need to add the new Kconfig option to ESP32-C3 defconfings(pm and autopm) to build the file and fix the error.

File: libs/libc/machine/Kconfig [hunk 1]
```diff
@@ -44,6 +44,10 @@ config ARCH_ROMGETC
 # Default settings for C library functions that may be replaced with
 # architecture-specific versions.
 
+config LIBC_ARCH_ATOMIC
```
Comment (gustavonihei) [inline libs/libc/machine/Kconfig:47]: The implementation for the atomic builtins should not be optional.
Instead, the OS-supported implementation should be used by every architecture that does not provide instructions for atomic memory operations.
We should create a hidden `bool` config (e.g. `ARCH_HAVE_ATOMICS`) and those ISA that do not provide (armv6-m, RISC-V without the `A` extension...) would be the ones to **not** select it.

File: boards/risc-v/esp32c3/esp32c3-devkit/configs/pm/defconfig [hunk 2]
```diff
@@ -23,6 +23,7 @@ CONFIG_DEV_ZERO=y
 CONFIG_FS_PROCFS=y
 CONFIG_IDLETHREAD_STACKSIZE=2048
 CONFIG_INTELHEX_BINARY=y
+CONFIG_LIBC_ARCH_ATOMIC=y
```
Comment (xiaoxiang781216) [inline boards/risc-v/esp32c3/esp32c3-devkit/configs/pm/defconfig:26]: let's select LIBC_ARCH_ATOMIC under ARCH_CHIP_ESP32C3 in arch/riscv/Kconfig

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
It should work now, please take a look, @Ouss4 and @gustavonihei .

**Review by gustavonihei — APPROVED** [review]

**Pull request merged**